service/controller/clusterapi/v29/key/machine_deployment_test.go

```diff
@@ -0,0 +1,159 @@
+package key
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+
+	g8sclusterv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	g8sproviderv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/aws-operator/pkg/annotation"
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func withg8sMachineDeploymentSpecToCMAMachineDeployment(cr v1alpha1.MachineDeployment, providerExtension g8sclusterv1alpha1.AWSMachineDeploymentSpec) v1alpha1.MachineDeployment {
+	var err error
+
+	if cr.Spec.Template.Spec.ProviderSpec.Value == nil {
+		cr.Spec.Template.Spec.ProviderSpec.Value = &runtime.RawExtension{}
+	}
+
+	cr.Spec.Template.Spec.ProviderSpec.Value.Raw, err = json.Marshal(&providerExtension)
+	if err != nil {
+		panic(err)
+	}
+
+	return cr
+}
+
+func TestStatusAvailabilityZones(t *testing.T) {
+	testCases := []struct {
+		name              string
+		cr                v1alpha1.MachineDeployment
+		providerExtension g8sclusterv1alpha1.AWSMachineDeploymentSpec
+		expectedZones     []g8sproviderv1alpha1.AWSConfigStatusAWSAvailabilityZone
+		errorMatcher      func(error) bool
+	}{
+		{
+			name: "case 0: Test with two AZs",
+			cr: v1alpha1.MachineDeployment{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						annotation.MachineDeploymentSubnet: "10.100.4.0/24",
+					},
+				},
+			},
+			providerExtension: g8sclusterv1alpha1.AWSMachineDeploymentSpec{
+				Provider: g8sclusterv1alpha1.AWSMachineDeploymentSpecProvider{
+					AvailabilityZones: []string{
+						"eu-central-1a",
+						"eu-central-1b",
+					},
+				},
+			},
+			expectedZones: []g8sproviderv1alpha1.AWSConfigStatusAWSAvailabilityZone{
+				{
+					Name: "eu-central-1a",
+					Subnet: g8sproviderv1alpha1.AWSConfigStatusAWSAvailabilityZoneSubnet{
+						Private: g8sproviderv1alpha1.AWSConfigStatusAWSAvailabilityZoneSubnetPrivate{
+							CIDR: "10.100.4.0/26",
```

Comment:
When `/24` network is split into two - why it isn't `/25` network?

Reply:
Because it's split into two and then other half of those is again split into two:

Cluster: `/24`
Availability Zone: `/25`
Public / Private on AZ: `/26`

Reply:
There are private and public for each AZ. OK that makes sense 👍